### PR TITLE
Port bookshelf to env:flex and Cloud SQL v2

### PR DESCRIPTION
**Pull request by ace-n**
@jmdobry @jonparrott PTAL

### CODE REVIEW

**Comment by ace-n** [comment]
@jmdobry Noted - working on that now.

**Comment by ace-n** [comment]
** Don't merge this yet ** - the associated documentation is still a WIP. Thanks!

**Comment by ace-n** [comment]
This should be good to merge now, FYI.

File: 2-structured-data/books/model-cloudsql.js [hunk 1]
```diff
@@ -18,13 +18,17 @@ const mysql = require('mysql');
...
-    host: config.get('MYSQL_HOST'),
+  const options = {
     user: config.get('MYSQL_USER'),
-    password: config.get('MYSQL_PASSWORD')
-  }));
+    password: config.get('MYSQL_PASSWORD'),
+    database: 'bookshelf'
+  };
+
+  if (process.env.INSTANCE_CONNECTION_NAME) {
```
Comment (jmdobry) [inline 2-structured-data/books/model-cloudsql.js]: Needs to be:

```js
if (process.env.INSTANCE_CONNECTION_NAME && config.get('NODE_ENV') === 'production') {
```

File: 3-binary-data/books/model-cloudsql.js [hunk 2]
```diff
@@ -18,13 +18,17 @@ const mysql = require('mysql');
...
-    host: config.get('MYSQL_HOST'),
+  const options = {
     user: config.get('MYSQL_USER'),
-    password: config.get('MYSQL_PASSWORD')
-  }));
+    password: config.get('MYSQL_PASSWORD'),
+    database: 'bookshelf'
+  };
+
+  if (process.env.INSTANCE_CONNECTION_NAME) {
```
Comment (jmdobry) [inline 3-binary-data/books/model-cloudsql.js]: Needs to be:

```js
if (process.env.INSTANCE_CONNECTION_NAME && config.get('NODE_ENV') === 'production') {
```

File: 4-auth/books/model-cloudsql.js [hunk 3]
```diff
@@ -18,13 +18,17 @@ const mysql = require('mysql');
...
-    host: config.get('MYSQL_HOST'),
+  const options = {
     user: config.get('MYSQL_USER'),
-    password: config.get('MYSQL_PASSWORD')
-  }));
+    password: config.get('MYSQL_PASSWORD'),
+    database: 'bookshelf'
+  };
+
+  if (process.env.INSTANCE_CONNECTION_NAME) {
```
Comment (jmdobry) [inline 4-auth/books/model-cloudsql.js]: Needs to be:

```js
if (process.env.INSTANCE_CONNECTION_NAME && config.get('NODE_ENV') === 'production') {
```

File: 5-logging/books/model-cloudsql.js [hunk 4]
```diff
@@ -18,13 +18,17 @@ const mysql = require('mysql');
...
-    host: config.get('MYSQL_HOST'),
+  const options = {
     user: config.get('MYSQL_USER'),
-    password: config.get('MYSQL_PASSWORD')
-  }));
+    password: config.get('MYSQL_PASSWORD'),
+    database: 'bookshelf'
+  };
+
+  if (process.env.INSTANCE_CONNECTION_NAME) {
```
Comment (jmdobry) [inline 5-logging/books/model-cloudsql.js]: Needs to be:

```js
if (process.env.INSTANCE_CONNECTION_NAME && config.get('NODE_ENV') === 'production') {
```

File: 6-pubsub/books/model-cloudsql.js [hunk 5]
```diff
@@ -19,13 +19,17 @@ const config = require('../config');
...
-    host: config.get('MYSQL_HOST'),
+  const options = {
     user: config.get('MYSQL_USER'),
-    password: config.get('MYSQL_PASSWORD')
-  }));
+    password: config.get('MYSQL_PASSWORD'),
+    database: 'bookshelf'
+  };
+
+  if (process.env.INSTANCE_CONNECTION_NAME) {
```
Comment (jmdobry) [inline 6-pubsub/books/model-cloudsql.js]: Needs to be:

```js
if (process.env.INSTANCE_CONNECTION_NAME && config.get('NODE_ENV') === 'production') {
```

File: 7-gce/books/model-cloudsql.js [hunk 6]
```diff
@@ -19,13 +19,17 @@ const config = require('../config');
...
-    host: config.get('MYSQL_HOST'),
+  const options = {
     user: config.get('MYSQL_USER'),
-    password: config.get('MYSQL_PASSWORD')
-  }));
+    password: config.get('MYSQL_PASSWORD'),
+    database: 'bookshelf'
+  };
+
+  if (process.env.INSTANCE_CONNECTION_NAME) {
```
Comment (jmdobry) [inline 7-gce/books/model-cloudsql.js]: Needs to be:

```js
if (process.env.INSTANCE_CONNECTION_NAME && config.get('NODE_ENV') === 'production') {
```

File: optional-container-engine/books/model-cloudsql.js [hunk 7]
```diff
@@ -19,13 +19,17 @@ const config = require('../config');
...
-    host: config.get('MYSQL_HOST'),
+  const options = {
     user: config.get('MYSQL_USER'),
-    password: config.get('MYSQL_PASSWORD')
-  }));
+    password: config.get('MYSQL_PASSWORD'),
+    database: 'bookshelf'
+  };
+
+  if (process.env.INSTANCE_CONNECTION_NAME) {
```
Comment (jmdobry) [inline optional-container-engine/books/model-cloudsql.js]: Needs to be:

```js
if (process.env.INSTANCE_CONNECTION_NAME && config.get('NODE_ENV') === 'production') {
```

**Pull request merged**